addons/pos_gift_card/static/src/js/GiftCardPopup.js

```diff
@@ -152,15 +152,56 @@ odoo.define("pos_gift_card.GiftCardPopup", function (require) {
 
       let currentOrder = this.env.pos.get_order();
       let lineUsed = await this.isGiftCardAlreadyUsed()
-      if (lineUsed) currentOrder.remove_orderline(lineUsed);
-
-      await currentOrder.add_product(gift, {
-        price: this.getPriceToRemove(giftCard),
-        quantity: 1,
-        merge: false,
-        gift_card_id: giftCard.id,
-        extras: { price_automatically_set: true },
-      });
+      if (lineUsed)
+        lineUsed.forEach(line => currentOrder.remove_orderline(line));
+
+      /**
+       * Obtain the total amount to pay that can be covered by a gift card (if
+       * the balance of the gift card allows to cover the total of the order
+       * will be the total of the order, otherwise will be the balance of the
+       * gift card) and the total of the order that includes all lines.
+       * */
+      let total_amount_to_pay = this.getPriceToRemove(giftCard);
+      let current_order_total = currentOrder.get_total_with_tax();
+      if (!current_order_total) return;
```

Comment:
Odoo does not have this return early. If the quantity is 0, it adds a 0 giftcard line.

Please remove that return early, so that behavior is kept.

Reply:
Right now with the changes, using as a base the code of promotions, we are only creating payments when the payment amount is a negative amount

Also, If I don't return early an error will be shown since I'm using that value to calculate the `basePercentage` and it will mean a division by 0